AlphaWallet/Core/Views/TableViewDiffableDataSource.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  TableViewDiffableDataSource.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 18.07.2022.
+//
+
+import UIKit
+import Combine
+
+class TableViewDiffableDataSource<SectionIdentifierType: Hashable, ItemIdentifierType: Hashable>: UITableViewDiffableDataSource<SectionIdentifierType, ItemIdentifierType> {
+    let numberOfRowsInSection: PassthroughSubject<Int, Never> = .init()
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        numberOfRowsInSection.send(section)
```

Comment:
This seems to rely on call order in the UITableView implemetation? I don't think this is a good idea. 

```
numberOfRowsInSection.send(section)
```

Changes in the UITableView implementation would break this (e.g caching etc).

Reply:
Tha comes from our imp, we update tableView empty view height constraint according to the call of function 
`func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int` but with diffable datasource we able to get this call only if we override DataSource

Reply:
Hmm.. Ok